compiler/haskell-ide-core/src/Development/IDE/Types/Location.hs

```diff
@@ -18,67 +18,13 @@ module Development.IDE.Types.Location
     , appendRange
     ) where
 
-import Control.DeepSeq  (NFData (..))
-import Data.Aeson.Types (FromJSON, FromJSONKey, ToJSON, ToJSONKey)
-import Data.Binary (Binary)
-import Data.Data
-import Data.Text.Prettyprint.Doc.Syntax
-import GHC.Generics
-
-------------------------------------------------------------------------------
---- Types
-------------------------------------------------------------------------------
-
--- | Position in a text document expressed as zero-based line and
---   character offset.
-data Position = Position
-    { positionLine      :: {-# UNPACK #-} !Int
-      -- ^ Zero-based line position in the document.
-    , positionCharacter :: {-# UNPACK #-} !Int
-      -- ^ Zero-based character offset on the line.
-    } deriving (Eq, Ord, Read, Show, Generic, Data)
-
-instance NFData Position
-
-instance Pretty Position where
-    pretty pos =
-        pretty (positionLine pos + 1) <> colon <> pretty (positionCharacter pos + 1)
-
-
--- | A range in a text document expressed as inclusive start-position and an
---   exclusive end-position.
-data Range = Range
-    { rangeStart :: {-# UNPACK #-} !Position
-      -- ^ The start position of the range, which is considered to be part of
-      -- the range.
-    , rangeEnd   :: {-# UNPACK #-} !Position
-      -- ^ The end position of the range, which is not considered to be part
-      -- of the range.
-    } deriving (Eq, Ord, Read, Show, Generic, Data)
-
-instance NFData Range
-
-instance Pretty Range where
-    pretty range =
-        pretty (rangeStart range) <> "-" <> pretty (rangeEnd range)
-
-
--- | Represents a location inside a resource, such as a line inside a text file.
-data Location = Location
-    { lFilePath :: !FilePath
-      -- ^ The uri of the document.
-    , lRange    :: !Range
-      -- ^ The range within the document.
-    } deriving (Eq, Ord, Read, Show, Generic, Data)
-
-instance NFData Location
-
+import "haskell-lsp-types" Language.Haskell.LSP.Types (Location(..), Range(..), Position(..), Uri(..))
```

Comment:
Why the package import?

Reply:
If we have a clash, let's move our custom version.

Reply:
This was useful, but it isn't any more